stdlib/public/runtime/Heap.cpp

```diff
@@ -93,15 +97,32 @@ void *swift::swift_slowAlloc(size_t size, size_t alignMask) {
     p = malloc(size);
 #endif
   } else {
-    size_t alignment = (alignMask == ~(size_t(0)))
-                           ? _swift_MinAllocationAlignment
-                           : alignMask + 1;
+    size_t alignment = computeAlignment(alignMask);
     p = AlignedAlloc(size, alignment);
   }
   if (!p) swift::crash("Could not allocate memory.");
   return p;
 }
 
+void *swift::swift_slowAllocTyped(size_t size, size_t alignMask,
+                                  MallocTypeId typeId) {
```

Comment:
At first, I had the calls to the old and new malloc APIs in one function and let the untyped one forward to the typed one:
```
void *swift_slowAlloc(size_t size, size_t alignMask) {
  return swift_slowAllocTyped(size, alignMask, /*typeId=*/0);
}
```
But the `#if` became very unwieldy, so I decided to turn it around (Typed variant doing it's thing and having the untyped one as a fallback) and accept a bit of code duplication.